client/src/extension.ts

```diff
@@ -53,29 +55,62 @@ export function activate(context: vscode.ExtensionContext) {
   // client can be deactivated on extension deactivation
   context.subscriptions.push(...registerCommands(client), client.start());
 
-  client.onDidChangeState((e) => {
-    let task: {resolve: () => void}|undefined;
-    if (e.newState == lsp.State.Running) {
-      client.onNotification(projectLoadingNotification.start, () => {
-        if (task) {
-          task.resolve();
-          task = undefined;
-        }
-        vscode.window.withProgress(
-            {
-              location: vscode.ProgressLocation.Window,
-              title: 'Initializing Angular language features',
-            },
-            () => new Promise((resolve) => {
-              task = {resolve};
-            }));
-      });
+  client.onDidChangeState((e: lsp.StateChangeEvent) => {
+    if (e.newState === lsp.State.Running) {
+      registerNotificationHandlers(client);
+    }
+  });
+}
 
-      client.onNotification(projectLoadingNotification.finish, () => {
-        if (task) {
-          task.resolve();
-          task = undefined;
-        }
+function registerNotificationHandlers(client: lsp.LanguageClient) {
+  let task: {resolve: () => void}|undefined;
+  client.onNotification(notification.ProjectLoadingStart, () => {
+    if (task) {
+      task.resolve();
+      task = undefined;
+    }
```

Comment:
nit
```suggestion
      task?.resolve();
      task = undefined;
```

Or even pull out into a function `closeLastTask`